src/flint/types/fmpz_mod_poly.pyx

```diff
@@ -279,11 +465,61 @@ cdef class fmpz_mod_poly(flint_poly):
         return res
 
     def is_irreducible(self):
-        pass
+        """
+        Return whether this polynomial is irreducible.
+        """
+        return 1 == fmpz_mod_poly_is_irreducible(self.val, self.ctx.mod.val)
 
     def is_squarefree(self):
-        pass
+        """
+        Return whether this polynomial is squarefree.
+        """
+        return 1 == fmpz_mod_poly_is_squarefree(self.val, self.ctx.mod.val)
+
+    def radical(self):
+        """
+        Return the radical of self, the product of the irreducible
+        factors of the polynomial. This is also referred to as the
+        square-free part of the polynomial.
+        """
+        res = self.ctx.one()
+        # TODO: this should be the squarefree leading coeff I guess...
+        res *= self.leading_coefficient()
+        _, factors = self.factor()
```

Comment:
Should this not use `factor_squarefree`?

Reply:
In fact is it not just `p / gcd(p, p')`?

Reply:
Yeah this function is wrong. It's a in progress function which I've abandoned but pushed as I'm off to do some non-computer stuff for a bit. Sorry. I should have kept it local